frontend/packages/moocfi-auth/index.ts

```diff
@@ -0,0 +1,181 @@
+import axios from "axios"
+import Cookies from "universal-cookie"
+
+/*
+const axios = require("axios")
+const Cookies = require("universal-cookie")
+*/
+
+const BASE_URL =
+  process.env.NODE_ENV === "production"
+    ? "https://mooc.fi"
+    : "http://localhost:4000"
+
+interface ExtraFields {
+  namespace?: string
+  data?: any
+}
+
+interface UserFields {
+  organizational_id?: string
+  first_name?: string
+  last_name?: string
+}
+
+interface CreateFields {
+  email: string
+  password: string
+  confirmPassword: string
+  extra_fields: ExtraFields
+  user_fields: UserFields
+  origin?: string
+  provider?: string
+  language?: string
+  domain?: string
+  priority?: string
+}
+
+export const createUser = async (data: CreateFields) => {
+  const cookies = new Cookies()
+  let origin = data.origin || "mooc.fi"
+  let provider = data.provider || "native"
+  let language = data.language || "en"
+  let domain = data.domain || "localhost"
+  let priority = data.priority || null
+
+  return await axios({
+    method: "POST",
+    url: `${BASE_URL}/auth/signUp`,
+    data: {
+      email: data.email,
+      password: data.password,
+      confirmPassword: data.confirmPassword,
+      extra_fields: data.extra_fields,
+      user_fields: data.user_fields,
+      provider,
+      origin,
+      language,
+    },
+    headers: { "Content-Type": "application/json" },
+  })
+    .then((response) => response.data)
```

Comment:
I would use async/await/try-catch everywhere since it's what our programmers are used to and then can sometimes be a bit unintuitive.